vllm/core/block_manager_v2.py

```diff
@@ -85,17 +98,31 @@ def __init__(
         )
 
         self.block_tables: Dict[SeqId, BlockTable] = {}
+        self.cross_block_tables: Dict[EncoderSeqId, BlockTable] = {}
 
     def can_allocate(self, seq_group: SequenceGroup) -> AllocStatus:
         # FIXME(woosuk): Here we assume that all sequences in the group share
         # the same prompt. This may not be true for preempted sequences.
-        seq = seq_group.get_seqs(status=SequenceStatus.WAITING)[0]
 
+        is_encoder_decoder = seq_group.is_encoder_decoder()
+        if self.enable_caching and is_encoder_decoder:
+            raise NotImplementedError(str_not_impl_enc_dec_prefix_cache)
+
+        if self.block_sliding_window is not None and is_encoder_decoder:
+            raise NotImplementedError(str_not_impl_enc_dec_swa)
```

Comment:
Same comment as above about single `if is_encoder_decoder`.

Reply:
Thanks! Fixed in most recent push.